newrelic-agent/src/main/java/com/newrelic/agent/TimedTokenSet.java

```diff
@@ -37,14 +36,13 @@ public TimedTokenSet(int timeOut, TimeUnit unit, final ExpirationService expirat
         // onRemoval happens immediately after put, so we can hit onRemoval logic before getToken() even finishes
         long timeOutMilli = TimeConversion.convertToMilliWithLowerBound(timeOut, unit, 250L);
 
-        activeTokens = CacheBuilder.newBuilder()
-                .concurrencyLevel(8)
+        activeTokens = Caffeine.newBuilder()
+                .initialCapacity(8)
                 .expireAfterAccess(timeOutMilli, TimeUnit.MILLISECONDS)
+                .executor(Runnable::run)
```

Comment:
This `Runnable::run` is here because one of the tests expects to see the effects of the removal notification in a timely manner. See the [wiki](https://github.com/ben-manes/caffeine/wiki/Guava) on how Caffeine may schedule this kind of work on another thread. Alternatively the test could be tweaked. If the current behavior is desired, perhaps `.executor(Runnable::run)` should also be added in `AsyncTransactionService`.